src/exec.js

```diff
@@ -58,19 +60,16 @@ function execSync(cmd, opts) {
   if (fs.existsSync(codeFile)) common.unlinkSync(codeFile);
 
   var execCommand = '"'+process.execPath+'" '+scriptFile;
-  var execOptions = {
-    env: process.env,
-    cwd: _pwd(),
-    maxBuffer: 20*1024*1024
-  };
-
+  opts.cwd = opts.cwd || _pwd();
+  opts.env = opts.env || process.env;
+  opts.maxBuffer = opts.maxBuffer || DEFAULT_MAXBUFFER_SIZE;
```

Comment:
`common.extend()` prefers new values over old values. This block prefers old values over new values. What I mean is...

If someone supplies a `cwd` option to `exec()`, it will not be overridden by `_pwd()`. If someone does not provide it, then it defaults to `_pwd()`. Same logic for the other options (a default maxbuffer value was provided before, so I kept that as well). `process.env` is good because it makes sure any `env` changes get passed through to the exec call.